server/executor/default_poller_executor.go

```diff
@@ -189,7 +194,7 @@ func (pe DefaultPollerExecutor) donePollingTraces(job *PollingRequest, traceDB t
 	if !traceDB.ShouldRetry() {
 		return true, "TraceDB is not retryable"
 	}
-	pp := *pe.ppGetter.GetDefault(job.ctx).Periodic
+	pp := *pe.ppGetter.GetDefault(job.Context()).Periodic
```

Comment:
since you're here, can you add a `nil` check for `Peridiodic`? It should never be nil, but it's better to be safe than panic